flytekit/extend/backend/agent_service.py

```diff
@@ -17,37 +19,47 @@
 from flytekit.models.task import TaskTemplate
 
 
-class AgentService(AsyncAgentServiceServicer):
-    def CreateTask(self, request: CreateTaskRequest, context: grpc.ServicerContext) -> CreateTaskResponse:
+class AsyncAgentService(AsyncAgentServiceServicer):
+    async def CreateTask(self, request: CreateTaskRequest, context: grpc.ServicerContext) -> CreateTaskResponse:
         try:
             tmp = TaskTemplate.from_flyte_idl(request.template)
             inputs = LiteralMap.from_flyte_idl(request.inputs) if request.inputs else None
             agent = AgentRegistry.get_agent(context, tmp.type)
             if agent is None:
                 return CreateTaskResponse()
-            return agent.create(context=context, inputs=inputs, output_prefix=request.output_prefix, task_template=tmp)
+            if agent.asynchronous:
+                return await agent.async_create(
```

Comment:
lets put a 
```python
log.debug()
try:
   agent.async
except Exception:
   log.error
   raise
log.debug(done)
```

Reply:
the idea is we want to log and know what went wrong in production. Think about how will you make it easier to debug in production and manage in production. Also we should think about adding prometheus metrics or open telemetry (separate PR is ok)

Reply:
done